File: src/main/java/de/johoop/testng/TestNGFramework.java
package de.johoop.testng;

import org.scalatools.testing.Fingerprint;
import org.scalatools.testing.Framework;
import org.scalatools.testing.Logger;
import org.scalatools.testing.Runner;

public class TestNGFramework implements Framework {

  private static Fingerprint[] FINGERPRINTS = new Fingerprint[] { new TestNGFingerprint() };
  
  @Override public String name() {
    return "TestNG";
  }

  @Override public Runner testRunner(final ClassLoader testClassLoader, final Logger[] loggers) {
    return new TestNGRunner(testClassLoader, loggers);
  }

  @Override public Fingerprint[] tests() {
    return FINGERPRINTS;
  }
}

File: src/main/scala/de/johoop/testnginterface/TestNGFramework.scala
package de.johoop.testnginterface

import org.scalatools.testing.Fingerprint
import org.scalatools.testing.Framework
import org.scalatools.testing.Logger

import java.util.concurrent.Semaphore

class TestNGFramework extends Framework {
  val name = "TestNG"
  def testRunner(testClassLoader: ClassLoader, loggers: Array[Logger]) = new TestNGRunner(testClassLoader, loggers)
  val tests = Array[Fingerprint](AllClassesFingerprint)
}